### PR TITLE
Adjustable clipping

### PR DESCRIPTION
**Pull request by OnTheWheel**
I added adjustable gradient clipping(AGC) to sgd_solver.
AGC is needed to shorten learning time during training deep CNN for super-resolution.
I've tested this code, and it worked well for my super-resolution task.

In short description, it adjusts clipping threshold based on current learning rate :
    threshold = clip_gradients / current_lr 

For more information, refer to the following paper :
   Accurate Image Super-Resolution Using Very Deep Convolutional Networks
  ( Jiwon Kim et al. )

### CODE REVIEW

**Comment by OnTheWheel** [comment]
@jeffdonahue Could you review this pull request?

**Comment by Soonhwan-Kwon** [comment]
please anyone can confirm this PR?
